### ISSUE TITLE
Type inference failed. This indicates an invalid graph that escaped type checking. Error message: INVALID_ARGUMENT

### ISSUE THREAD

**Issue by sky712345678** [keras-team/keras#16978]
**System information**.
- Have I written custom code (as opposed to using a stock example script provided in Keras): no
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Linux Ubuntu 18.04
- TensorFlow installed from (source or binary): binary
- TensorFlow version (use command below): 2.9
- Python version: 3.7
- Bazel version (if compiling from source): N/A
- GPU model and memory: N/A (CPU)
- Exact command to reproduce:
1. Open the [notebook](https://drive.google.com/file/d/1k78lpGVthB7nthEkYgUs3JNJTuR79r5E/view?usp=sharing) with Google Colab
2. Run all cells
3. View the runtime logs
![image](https://user-images.githubusercontent.com/37875281/188081070-58f6db7c-8cfe-4c0e-90c3-926cc2e4261b.png)
Note: We have to upgrade Tensorflow and Keras to 2.9 manually in the notebook, because the current default version is not the latest one on Colab.

**Describe the problem**.
(Continue the issue from [tensorflow_issue_57052](https://github.com/tensorflow/tensorflow/issues/57052#issuecomment-1234440360))
I got a `Type inference failed` error when running `tf.keras.Model.fit()` in Tensorflow 2.9 and Keras 2.9. I didn't see this kind of error in version 2.8 with the identical code.
Although the program didn't crash, I'm afraid that there will be some error in the trained model.

**Describe the current behavior**.
Run `tf.keras.Model.fit()` and the error `Type inference failed` shows up.

**Describe the expected behavior**.
The error shouldn't show up.

**[Contributing](https://github.com/keras-team/keras/blob/master/CONTRIBUTING.md)**.

- Do you want to contribute a PR? (yes/no): no
- If yes, please read [this page](https://github.com/keras-team/keras/blob/master/CONTRIBUTING.md) for instructions
- Briefly describe your candidate solution(if contributing):

**Standalone code to reproduce the issue**.
Link to notebook: https://drive.google.com/file/d/1k78lpGVthB7nthEkYgUs3JNJTuR79r5E/view?usp=sharing
To reproduce:
1. Open the notebook with Google Colab
2. Run all cells
3. View the runtime logs

**Source code / logs**.
```shell
2022-08-20 17:18:05.533157: W tensorflow/core/common_runtime/forward_type_inference.cc:231] Type inference failed. This indicates an invalid graph that escaped type checking. Error message: INVALID_ARGUMENT: expected compatible input types, but input 1:
type_id: TFT_OPTIONAL
args {
  type_id: TFT_PRODUCT
  args {
    type_id: TFT_TENSOR
    args {
      type_id: TFT_BOOL
    }
  }
}
 is neither a subtype nor a supertype of the combined inputs preceding it:
type_id: TFT_OPTIONAL
args {
  type_id: TFT_PRODUCT
  args {
    type_id: TFT_TENSOR
    args {
      type_id: TFT_LEGACY_VARIANT
    }
  }
}

        while inferring type of node 'dice_loss/cond/output/_11'
```

**Comment by sushreebarsa** [comment]
@gadagashwini I was able to replicate the issue on colab, please find the [gist](https://colab.research.google.com/gist/sushreebarsa/720ad0e6f1304d4002e2f4fe92c6acfd/3d_unet_train_type_inference_failed.ipynb#scrollTo=kUprF0LhAl_J) here. Thank you!

**Comment by gadagashwini** [comment]
Hi @sky712345678, 
`W tensorflow/core/common_runtime/forward_type_inference.cc:231] Type inference failed.` is just a warning, you can safely ignore it. Given code executed without any error message. Thank you!

**Comment by tgsmith61591** [comment]
@gadagashwini what's the point of a warning if the response is simply `you can safely ignore it.`? It's clearly there for a reason

**Comment by sky712345678** [comment]
@gadagashwini can you talk a little bit more about the reason why we can safely ignore it? Thank you!

**Comment by gowthamkpr** [comment]
@sky712345678 This looks like an issue from tensorflow. Can you please create this issue in [tensorflow/tensorflow](https://github.com/tensorflow/tensorflow/issues/new/choose). Thank you!!

**Comment by foxik** [comment]
@gowthamkpr Well, the problem was first reported in tensorflow as https://github.com/tensorflow/tensorflow/issues/57052, but the guys there told the reporter to instead post an issue here.

If you know any more details (why it is a TensorFlow issue), could you please provide more details that we can give to the TF guys?

**Comment by foxik** [comment]
@gowthamkpr The issue was originally a TF issue, but we were redirected to post it here. If you know any more details (why it is a TF issue and not Keras), could you please provide more details that we can give to the TF guys? Thanks!

**Comment by zrx563010758** [comment]
Unsubscribe



---Original---
From: "Milan ***@***.***&gt;
Date: Sun, Oct 2, 2022 16:54 PM
To: ***@***.***&gt;;
Cc: ***@***.***&gt;;
Subject: Re: [keras-team/keras] Type inference failed. This indicates aninvalid graph that escaped type checking. Error message: INVALID_ARGUMENT(Issue #16978)




 
@gowthamkpr Well, the problem was first reported in tensorflow as tensorflow/tensorflow#57052, but the guys there told the reporter to instead post an issue here.
 
If you know any more details (why it is a TensorFlow issue), could you please provide more details that we can give to the TF guys?

**Comment by fchollet** [comment]
The issue is at the level of the `dice_loss`. Can you try producing a reproduction script that only involves the loss function? Maybe just try to backprop through the loss function and see what happens.

I think this should be reproducible without involving any Keras logic, at which point the TF folks will definitely look at it. But anyway, as said before, this is just a warning, not something critical. You can ignore it.

**Comment by sky712345678** [comment]
Ok, I got it. Thank you!

I wasn't sure how to reproduce it only involving the loss function, this is my try:
https://colab.research.google.com/drive/1qxamrOaOqfVANzMnN-u--Sue4iPtJCtf?usp=sharing
![image](https://user-images.githubusercontent.com/37875281/196616317-62db8599-5e9c-4e99-878f-9390aa808c22.png)
Running this Colab notebook, I didn't see the error message in runtime logs.

**Comment by TuanHAnhVN** [comment]
> Ok, I got it. Thank you!
> 
> I wasn't sure how to reproduce it only involving the loss function, this is my try: https://colab.research.google.com/drive/1qxamrOaOqfVANzMnN-u--Sue4iPtJCtf?usp=sharing ![image](https://user-images.githubusercontent.com/37875281/196616317-62db8599-5e9c-4e99-878f-9390aa808c22.png) Running this Colab notebook, I didn't see the error message in runtime logs.

Hi bro,
Can you share the solution to over this issue?
I have met the similar problem.
Thank you so much

**Comment by isohrab** [comment]
I'm getting the same warning with TF 2.11 when I set mask_zero=True in the embedding layer.

**Comment by alibahmanyar** [comment]
+1;
I'm also getting the same warning with TF 2.11 and setting `mask_zero=True` in the embedding layer while training on GPU. After the warning, the model keeps training and is then saved, but the saved model can't be loaded using `keras.models.load_model`. 
However, when I'm training on CPU (even with `mask_zero=True`) everything works fine and the warning doesn't show up; the model is trained, saved and can be loaded and used again without encountering any problem.

**Comment by albertz** [comment]
I'm getting something very similar but with pure TF 2.11 on Mac M1. So I really think this is a pure TF issue, and we should reopen the TF issue (https://github.com/tensorflow/tensorflow/issues/57052).

**Comment by NinaCilliers** [comment]
I have the same issue unfortunately. Currently running with mask_zero set to True and using GPU.
